quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -454,6 +454,11 @@ async fn download_all(byte_stream: &mut ByteStream, output: &mut Vec<u8>) -> io:
 
 #[async_trait]
 impl Storage for S3CompatibleObjectStorage {
+    async fn check(&self) -> anyhow::Result<()> {
+        self.s3_client.list_buckets().await?;
```

Comment:
`list_buckets` is not a great candidate for a connectivity check. Quickwit does not require the `ListAllBuckets` permission to work correctly and most large organizations won't give applications the right to list all their buckets.

We could go with `list_objects` limited to 1 result or `delete_object` with a very random key.

Reply:
Ok, that makes sense.

Reply:
I changed it to `list_objects` with a `max_keys: Some(1)`.